core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/buffer.go

```diff
@@ -62,6 +103,27 @@ func (b fetchedBlock) Clone() fetchedBlock {
 	}
 }
 
+// Sort by log identifiers, shuffled using a pseduorandom souce that is shared across all nodes
+// for a given block.
+func (b *fetchedBlock) Sort() {
+	randSeed := random.GetRandomKeySource(nil, uint64(b.blockNumber))
+
+	shuffledLogIDs := make([]string, len(b.logs))
+	for i, log := range b.logs {
+		ext := ocr2keepers.LogTriggerExtension{
+			TxHash:    log.log.TxHash,
+			Index:     uint32(log.log.LogIndex),
+			BlockHash: log.log.BlockHash,
+		}
+		logID := hex.EncodeToString(ext.LogIdentifier())
+		shuffledLogIDs[i] = random.ShuffleString(logID, randSeed)
+	}
+
+	sort.Slice(b.logs, func(i, j int) bool {
+		return shuffledLogIDs[i] < shuffledLogIDs[j]
+	})
```

Comment:
just want to verify this will work properly if the indices passed in comparator doesn't refer to the slice being sorted (instead a constant slice)?

Reply:
Yes, the comparator accepts indices and returns a bool, making it completely decoupled from the underlying slice.